RadixWallet/Features/CreateAccount/Coordinator/CreateAccountCoordinator+Reducer.swift

```diff
@@ -176,124 +154,78 @@ extension CreateAccountCoordinator {
-			// We need to create the Profile before deriving the public key
+			// We need to create the Profile before creating the Account
 			.run { send in
 				try await onboardingClient.createNewProfile()
-				await send(.internal(.handleProfileCreated(factorSourceOption: factorSourceOption)))
+				await send(.internal(.handleProfileCreated(option)))
 			} catch: { error, _ in
 				errorQueue.schedule(error)
 			}
 		} else {
-			// We can derive the public key since the Profile has been created already
-			derivePublicKey(state: &state, factorSourceOption: factorSourceOption)
+			// We can create the Account since the Profile has been created already
+			createAccount(state: &state, option: option)
 		}
 	}
 
-	private func derivePublicKey(state: inout State, factorSourceOption: DerivePublicKeys.State.FactorSourceOption) -> Effect<Action> {
-		state.destination = .derivePublicKey(
-			.init(
-				derivationPathOption: .next(
-					for: .account,
-					networkID: state.config.specificNetworkID,
-					curve: .curve25519,
-					scheme: .cap26
-				),
-				factorSourceOption: factorSourceOption,
-				purpose: .createNewEntity(kind: .account)
-			))
-		return .none
+	private func createAccount(state: inout State, option: Option) -> Effect<Action> {
+		guard let name = state.name else {
+			fatalError("Name should be set before creating Account")
+		}
+		let displayName = DisplayName(nonEmpty: name)
+		return .run { [networkId = state.config.specificNetworkID] send in
+			let account = switch option {
+			case .bdfs:
+				try await SargonOS.shared.createAccountWithBDFS(networkId: networkId, name: displayName)
+			case let .specific(factorSource):
+				try await SargonOS.shared.createAccount(factorSource: factorSource, networkId: networkId, name: displayName)
+			}
+
+			let updated = await getThirdPartyDepositSettings(account: account)
+			// TODO: Remove once this is implemented in Sargon (https://radixdlt.atlassian.net/browse/ABW-4147)
+			try? await SargonOS.shared.updateAccount(updated: updated)
+			await send(.internal(.handleAccountCreated(updated)))
+		} catch: { error, _ in
+			errorQueue.schedule(error)
+		}
+	}
+
+	private func getThirdPartyDepositSettings(account: Account) async -> Account {
+		do {
+			if let updated = try await doAsync(
+				withTimeout: .seconds(5),
+				work: { try await onLedgerEntitiesClient.syncThirdPartyDepositWithOnLedgerSettings(account: account) }
+			) {
+				loggerGlobal.notice("Used OnLedger ThirdParty Deposit Settings")
+				return updated
+			} else {
+				return account
+			}
+		} catch {
+			loggerGlobal.notice("Failed to get OnLedger state for newly created account: \(account). Will add it with default third party deposit settings...")
+			return account
+		}
 	}
 }
 
-extension CreateAccountCoordinator.State {
-	var lastStepState: CreateAccountCoordinator.Path.State? {
-		path.last
+// MARK: CreateAccountCoordinator.Option
+extension CreateAccountCoordinator {
+	enum Option: Sendable, Hashable {
```

Comment:
I think we usually call this `Mode` ? Feels awfully close to `Optional` to call `Option`?